dashboard/src/deployment/DeploymentDetails.js

```diff
@@ -0,0 +1,88 @@
+import React, { Component } from 'react';
+import { apiGet } from '../api/api.js';
+import { Accordion, Header, Icon, List, Segment } from 'semantic-ui-react';
+import Loading from '../util/Loading.js';
+//import CommandInstruction from '../util/CommandInstruction.js';
+
+const MemberGroupsView = ({member_groups}) => (
+  <div>
+    {member_groups.map((item) => <MemberListComponent 
+      group={item.group}
+      members={item.members}
+    />)}
+  </div>
+);
+
+const MemberListView = ({group, activeMemberID, onClick, members}) => (
+  <Segment>
+    <Header>{group}</Header>
+    <List divided>
+      {members.map((item) => <MemberView memberInfo={item} active={item.id === activeMemberID} onClick={onClick}/>)}
+    </List>
+  </Segment>
+);
+
+const MemberView = ({memberInfo, active, onClick}) => (
+  <List.Item>
+    <Accordion>
+      <Accordion.Title active={active} onClick={() => onClick(memberInfo.id)}>
+        <Icon name='dropdown' /> {memberInfo.id}
+      </Accordion.Title>
+      <Accordion.Content active={active}>
+        <div>Pod: {memberInfo.pod_name}</div>
+        <div>PVC: {memberInfo.pvc_name}</div>
+        <div>PV: {memberInfo.pv_name}</div>
+      </Accordion.Content>
+    </Accordion>
+  </List.Item>
+);
+
+class MemberListComponent extends Component {
+  state = {};
+
+  onClick = (id) => { 
+    this.setState({activeMemberID:(this.state.activeMemberID === id) ? null : id}); 
```

Comment:
When using `this.state` to set state, you should use the functional form of `setState`:

```js
this.setState(({activeMemberID}) => ({
  activeMemberID: activeMemberID === id ? null : id
}));
```

Using `this.state` to derive new state is error-prone as `setState` is potentially async.

Reply:
BTW, I'd call the method "handleClick" so you pass `onClick={this.handleClick}` but I honestly have no idea where that convention comes from.